### PR TITLE
Adjust the serving threads to have a minimum of  threads

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
The current logic of determining the number of http serving threads has difficulties when the number of available cores are very small(like 1). this pr imposes a min of  threads in such scenarios to improve user experience.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 1]
```diff
@@ -146,7 +146,7 @@
             doc = "Number of threads to use for HTTP requests processing"
                 + " Default is set to `2 * Runtime.getRuntime().availableProcessors()`"
         )
-    private int numHttpServerThreads = 2 * Runtime.getRuntime().availableProcessors();
+    private int numHttpServerThreads = Math.max(4, 2 * Runtime.getRuntime().availableProcessors());
```
Comment (ivankelly) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: Is 4 enough? I've seen cases were I would still complain at 4.

Comment (srkukarni) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: What do you think would be the right number?
One thing is that with the current settings, on a one core machine, pulsar standalone wouldnt even come up. This fix solves that.

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run integration tests

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java (hunk 1 above)
Comment (ivankelly) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: I've been putting it at 16 to get around that issue, though admittedly I pulled that number out of thin air.

**Comment by srkukarni** [comment]
run java8 tests
run integration tests

**Comment by srkukarni** [comment]
run integration tests
run java8 tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
2f42dfd

**Pull request merged**